filebeat/docs/how-filebeat-works.asciidoc

```diff
@@ -21,60 +30,58 @@ Closing a harvester has the following consequences:
 
-The following example configures Filebeat to harvest lines from all log files that match the specified glob patterns:
+The following example configures {beatname_uc} to harvest lines from all log files that match the specified glob patterns:
 
-[source,yaml]
+["source","yaml",subs="attributes"]
 -------------------------------------------------------------------------------------
-filebeat.prospectors:
+{beatname_lc}.inputs:
 - type: log
   paths:
     - /var/log/*.log
     - /var/path2/*.log
 -------------------------------------------------------------------------------------
 
-Filebeat currently supports <<prospector-type,several `prospector` types>>. Each prospector type can be defined multiple times. The `log` prospector checks each file to see whether a harvester needs to be started, whether one is already running, or whether the file can be ignored (see <<ignore-older,`ignore_older`>>). New lines are only picked up if the size of the file has changed since the harvester was closed.
-
-NOTE: Filebeat prospectors can only read local files. There is no functionality to connect to remote hosts to read stored files or logs.
+Filebeat currently supports <<input-type,several `input` types>>. Each input type can be defined multiple times. The `log` input checks each file to see whether a harvester needs to be started, whether one is already running, or whether the file can be ignored (see <<ignore-older,`ignore_older`>>). New lines are only picked up if the size of the file has changed since the harvester was closed.
 
 [float]
-=== How does Filebeat keep the state of files?
+=== How does {beatname_uc} keep the state of files?
 
-Filebeat keeps the state of each file and frequently flushes the state to disk in the registry file. The state is used to remember the last offset a harvester was reading from and to ensure all log lines are sent. If the output, such as Elasticsearch or Logstash, is not reachable, Filebeat keeps track of the last lines sent and will continue reading the files as soon as the output becomes available again. While Filebeat is running, the state information is also kept in memory by each prospector. When Filebeat is restarted, data from the registry file is used to rebuild the state, and Filebeat continues each harvester at the last known position.
+{beatname_uc} keeps the state of each file and frequently flushes the state to disk in the registry file. The state is used to remember the last offset a harvester was reading from and to ensure all log lines are sent. If the output, such as Elasticsearch or Logstash, is not reachable, {beatname_uc} keeps track of the last lines sent and will continue reading the files as soon as the output becomes available again. While {beatname_uc} is running, the state information is also kept in memory for each input. When {beatname_uc} is restarted, data from the registry file is used to rebuild the state, and {beatname_uc} continues each harvester at the last known position.
```

Comment:
The 'flush' is the operation. One can change this to:
filebeat keeps the file states in memory and writes a snapshot to the registry file  whenever events are acknowledged by the outputs or when the registry flush timer is trigger. The flush timer is only active if some state has been changed.

Reply:
@urso I think you're proposing that I replace the entire line with your description, right?